packages/mui-base/src/Switch/Switch.test.tsx

```diff
@@ -1,78 +1,300 @@
 import * as React from 'react';
-import { createMount, createRenderer } from '@mui/internal-test-utils';
 import { expect } from 'chai';
-import { Switch, SwitchOwnerState, switchClasses } from '@mui/base/Switch';
-import { describeConformanceUnstyled } from '../../test/describeConformanceUnstyled';
+import { spy } from 'sinon';
+import { createRenderer, act } from '@mui/internal-test-utils';
+import { Switch } from '@mui/base/Switch';
 
 describe('<Switch />', () => {
-  const mount = createMount();
   const { render } = createRenderer();
 
-  describeConformanceUnstyled(<Switch />, () => ({
-    inheritComponent: 'span',
-    render,
-    mount,
-    refInstanceof: window.HTMLSpanElement,
-    testComponentPropWith: 'span',
-    slots: {
-      root: {
-        expectedClassName: switchClasses.root,
-      },
-      thumb: {
-        expectedClassName: switchClasses.thumb,
-      },
-      input: {
-        testWithElement: 'input',
-        expectedClassName: switchClasses.input,
-      },
-      track: {
-        expectedClassName: switchClasses.track,
-        isOptional: true,
+  // TODO: move to the conformance test suite
+
+  describe('prop: className', () => {
+    it('should apply the className when passed as a string', () => {
+      const { container } = render(<Switch className="test-class" />);
+      expect(container.firstElementChild?.className).to.contain('test-class');
+    });
+  });
+
+  describe('extra props', () => {
+    it('should spread extra props', () => {
+      const { container } = render(<Switch data-extra-prop="Lorem ipsum" />);
+      expect(container.firstElementChild as HTMLElement).to.have.attribute(
+        'data-extra-prop',
+        'Lorem ipsum',
+      );
+    });
+
+    it('should not override the built-in attributes', () => {
+      const { container } = render(<Switch data-state="checked" role="checkbox" />);
+      expect(container.firstElementChild as HTMLElement).to.have.attribute('role', 'switch');
+    });
```

Comment:
I believe users of Base UI should always be able to override the built-in attributes as an escape hatch, so their props should come last. Users of wrapped APIs operate at a higher level and shouldn't have the risk though. What do you think?

Reply:
Yeah, I guess it would make the components even more customizable and we still merge event handlers, so the likelihood of shooting yourself in the foot by adding more props is quite low. I'll update it.